src/app/dynamic/enterprise/quotas/service.ts

```diff
@@ -70,20 +75,25 @@ export class QuotaService {
   }
 
   getProjectQuota(projectId: string): Observable<QuotaDetails> {
-    if (!this.quotaMap.get(projectId)) {
+    if (!this.quotaMap.has(projectId)) {
       const quota$ = this._refreshTimer$.pipe(
         switchMap(_ =>
           this._http
             .get<QuotaDetails>(`${this._newRestRoot}/projects/${projectId}/quota`)
             .pipe(catchError(_ => of(null)))
         ),
-        shareReplay({refCount: true, bufferSize: 1})
+        shareReplay({refCount: true, bufferSize: 1}),
+        tap(quota => {
+          if (quota) {
+            this._previousQuotaMap.set(projectId, quota);
+          }
+        })
       );
 
       this.quotaMap.set(projectId, quota$);
     }
 
-    return this.quotaMap.get(projectId);
+    return this.quotaMap.get(projectId).pipe(startWith(this._previousQuotaMap.get(projectId) ?? null));
```

Comment:
Are we handling this `null` value where we call this method?

Reply:
Yes, I've put in `filter( Boolean)` where ever I'm. subscribing to quotas
<img width="587" alt="image" src="https://user-images.githubusercontent.com/56511988/189643339-3e9b8ec1-4cc1-42f6-913a-241a5d7dd27e.png">

Reply:
Yes, isLoading is being set to false instantly, I'm catering for this.